### PR TITLE
fix(SO): Add a search option to retrieve the latest follow-up, tasks, and solution

### PR DESCRIPTION
**Pull request by MyvTsv**
- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.

## Description

- It fixes !35024
- Add a search option to retrieve the date of the ticket's latest follow-up, tasks, and solution

## Screenshots (if appropriate):

Capture d’écran du 2024-11-12 17-13-59

### CODE REVIEW

File: src/Ticket.php [hunk 1]
```diff
@@ -3438,9 +3438,9 @@ public function rawSearchOptions()
             ]
         ];
 
-        $tab = array_merge($tab, ITILFollowup::rawSearchOptionsToAdd());
+        $tab = array_merge($tab, ITILFollowup::rawSearchOptionsToAdd('Ticket'));
```
Comment (trasher) [inline src/Ticket.php:3441]:
```suggestion
        $tab = array_merge($tab, ITILFollowup::rawSearchOptionsToAdd(self::class));
```

Comment (trasher) [reply src/Ticket.php:3441]: Try to use class name everywhere instead of hardcoded name

**Review by trasher — COMMENTED** [review]
Not tested, but seems correct

File: src/CommonITILObject.php [hunk 2]
```diff
@@ -4359,6 +4359,22 @@ public function getSearchOptionsSolution()
             ]
         ];
 
+        if ($itemtype === 'Ticket') {
```
Comment (trasher) [inline src/CommonITILObject.php:4362]:
```suggestion
        if ($itemtype === Ticket::class) {
```

File: src/CommonITILTask.php [hunk 3]
```diff
@@ -993,6 +995,23 @@ public static function rawSearchOptionsToAdd($itemtype = null)
             ]
         ];
 
+        if ($itemtype === 'Ticket') {
```
Comment (trasher) [inline src/CommonITILTask.php:998]:
```suggestion
        if ($itemtype === Ticket::class) {
```

File: src/ITILFollowup.php [hunk 4]
```diff
@@ -713,6 +715,23 @@ public static function rawSearchOptionsToAdd($itemtype = null)
             ]
         ];
 
+        if ($itemtype === 'Ticket') {
```
Comment (trasher) [inline src/ITILFollowup.php:718]:
```suggestion
        if ($itemtype === Ticket::class) {
```

**Review by trasher — APPROVED** [review]

File: src/CommonITILObject.php [hunk 5]
```diff
@@ -4272,7 +4272,7 @@ public function getSearchOptionsMain()
     /**
      * @since 0.85
      **/
-    public function getSearchOptionsSolution()
+    public function getSearchOptionsSolution($itemtype = null)
```
Comment (trasher) [inline src/CommonITILObject.php:4275]:
```suggestion
    public function getSearchOptionsSolution(?string $itemtype = null)
```

**Review by cedric-anne — COMMENTED** [review]
It seems OK, but why only adding these new search options only for tickets ? IMHO, all the ITIL objects having followup, solutions and task should get these new search options.

**Comment by trasher** [comment]
Indeed, it may be added for other related itemtypes. Could you try that way?

File: src/CommonITILObject.php [hunk 6]
```diff
@@ -4359,6 +4359,22 @@ public function getSearchOptionsSolution()
             ]
         ];
 
+        if ($itemtype === Ticket::class || $itemtype === Change::class || $itemtype === Problem::class) {
```
Comment (cedric-anne) [inline src/CommonITILObject.php:4362]: This condition should be removed.

```suggestion
```

File: src/CommonITILTask.php [hunk 7]
```diff
@@ -993,6 +995,23 @@ public static function rawSearchOptionsToAdd($itemtype = null)
             ]
         ];
 
+        if ($itemtype === Ticket::class || $itemtype === Change::class || $itemtype === Problem::class) {
```
Comment (cedric-anne) [inline src/CommonITILTask.php:998]:
```suggestion
```

File: src/ITILFollowup.php [hunk 8]
```diff
@@ -713,6 +715,23 @@ public static function rawSearchOptionsToAdd($itemtype = null)
             ]
         ];
 
+        if ($itemtype === Ticket::class || $itemtype === Change::class || $itemtype === Problem::class) {
```
Comment (cedric-anne) [inline src/ITILFollowup.php:718]:
```suggestion
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/CommonITILObject.php (hunk 6 above)
Comment (trasher) [reply src/CommonITILObject.php:4362]: Along with removing those conditions, you can also drop all the `get_class($this)` you added.

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**